notebooks/MixedModelsTutorial_basics.jl

```diff
@@ -66,124 +71,207 @@ Number of scores: 525126
+`PowerLOW` = power of lower limbs (i.e., standing long jump test), `PowerUP` = 
+apower of upper limbs (i.e., ball push test), SD = standard deviation. Points 
+are binned observed child means; lines are simple regression fits to the observations.
 """
 
 # ╔═╡ 58d561e2-77c0-43c2-990d-008286de4e5c
-md"""#### 1.3.1 Compute means"""
+md"""#### 1.3.2 Compute means
+The means scattered in Figures 1 and 2 are based on age bins of boys' and girls'  test performances. The means are stored in data frames `df2` for the full and `dat2` for the subset.
+"""
+
+# ╔═╡ f80c6440-6bf0-4656-97f5-15b547d724da
+begin
+df2 = combine(
+	groupby(
+		select(df, :, :age => ByRow(x -> round(x; digits=1)) => :age),
+		[:Sex, :Test, :age],
+	),
+	:zScore => mean => :zScore,
+	:zScore => length => :n,
+)
+end
+
+# ╔═╡ a450b88b-9ac4-4d44-9560-9e6d2e0dd8fa
+begin
+	design1 = mapping(:age, :zScore; color = :Sex, col = :Test)
+	lines1 = design1 * linear()
+	means1 = design1 * visual(Scatter, markersize=5)
+	data(df2) * means1 + data(df) * lines1 |> draw
+end
 
 # ╔═╡ e9280e08-ace4-48cd-ad4a-55501f315d6a
-dat2 = groupby(   # summary grouped data frame by test, sex, rounded age
-	 combine(
-		groupby(
-			select(dat,
-				:age => (x -> cut(x, 8)) => :Age,
-				:Sex,
-				:Test,
-				:zScore,
-				:age
-			),
-			[:Age, :Sex, :Test]),
-		:zScore => mean => :zScore,
-		:age => mean => :ageM
-		),
-	:Test
+begin
+dat2 = combine(
+	groupby(
+		select(dat, :, :age => ByRow(x -> round(x; digits=1)) => :age),
+		[:Sex, :Test, :age],
+	),
+	:zScore => mean => :zScore,
+	:zScore => length => :n,
 )
+end
+
+# ╔═╡ 462bd92a-056a-4e5d-8fa6-215784e0fb9d
+begin
+	design2 = mapping(:age, :zScore; color = :Sex, col = :Test)
+	lines2 = design2 * linear()
+	means2 = design2 * visual(Scatter, markersize=5)
+	data(dat2) * means2 + data(dat) * lines2 |> draw
+end
 
 # ╔═╡ 2574cf1c-4a9a-462b-a2f2-d599a8b42ec1
 md"""
 #### 1.3.2 Regression on `age` by `Sex` for each `Test`
-We compute the simple regression for each test for boys and girls using all observations.
+Another set of relevant statistics are the slopes for the regression of performance on age for boys and girls in each of the five tests. The lines in Figures 1 and 2, however, are computed directly from the raw data with the `linear()` command. 
 """
 
-# ╔═╡ a7e68776-7a42-4b4a-b540-f26b8b57d520
+# ╔═╡ 67b42d3f-0afe-4d16-817d-10212f5a5c90
 begin
-	test_names = levels(dat.Test)
-    comp_names = [     # establish order and labels of tbl.Test
-	    "Run" => "Endurance",
-	    "Star_r" => "Coordination",
-	    "S20_r" => "Speed",
-	    "SLJ" => "PowerLOW",
-	    "BPT" => "PowerUP"
-    ]
-end;
+    lms1 = DataFrame( 
+	    Sex = repeat(["Boys", "Girls"], inner=5), 
+	    Test = repeat(levels(df.Test), outer=2),
+	    GM = .0,
+		age = .0,
+		Estimate = "age_Sex_Text",
+	)
+  	for i in 1:10
```

Comment:
Is there a reason for building the dummy frame and looping over the rows instead of using a `groupby/combine`?

Reply:
I adopted this from one our other scripts